src/iplant_groups/clients/grouper.clj

```diff
@@ -484,32 +484,42 @@
 
 ;; This is only available as a Lite request; ActAsSubject works differently.
 (defn- format-group-folder-privileges-lookup-request
-  [entity-type username group-or-folder-name]
-  (if-let [name-key (get {:group :groupName
+  [entity-type username group-or-folder-name params]
+  (if-let [name-key (get {:group  :groupName
                           :folder :stemName}
                          entity-type)]
     {:WsRestGetGrouperPrivilegesLiteRequest
-     {:actAsSubjectId username
-      :includeSubjectDetail "T"
-      :includeGroupDetail   "T"
-      name-key group-or-folder-name}}
+     (remove-vals nil? {:actAsSubjectId       username
+                        :includeSubjectDetail "T"
+                        :includeGroupDetail   "T"
+                        name-key              group-or-folder-name
+                        :subjectId            (:subject-id params)
+                        :subjectSourceId      (:subject-source-id params)
+                        :privilegeName        (:privilege params)})}
     (throw+ {:type :clojure-commons.exception/bad-request :entity-type entity-type})))
 
+(defn- filter-privileges
+  [privileges {:keys [subject-source-id]}]
+  (if subject-source-id
+    (filter (fn [priv] (= (:sourceId (:wsSubject priv)) subject-source-id)) privileges)
+    privileges))
+
 (defn- get-group-folder-privileges
-  [entity-type username group-or-folder-name]
+  [entity-type username name params]
   (with-trap [default-error-handler]
-    (let [response (-> (format-group-folder-privileges-lookup-request entity-type username group-or-folder-name)
+    (let [response (-> (format-group-folder-privileges-lookup-request entity-type username name params)
                        (grouper-post "grouperPrivileges")
                        :WsGetGrouperPrivilegesLiteResult)]
-      [(:privilegeResults response) (:subjectAttributeNames response)])))
+      [(filter-privileges (:privilegeResults response) params)
+       (:subjectAttributeNames response)])))
```

Comment:
I shortened the `group-or-folder-name` parameter in order to avoid exceeding 120 characters per line without having to introduce newlines.